xxhash 0.8.1 fails to compile on RHEL 7 ppc64le

The compilation error is:
```
In file included from xxhash.c:43:0:
xxhash.h: In function 'XXH3_accumulate_512_vsx':
xxhash.h:4148:9: error: invalid parameter combination for AltiVec intrinsic
         xxh_u64x2 acc_vec        = vec_xl(0, xacc + 2 * i);
         ^
xxhash.h:4158:9: error: invalid parameter combination for AltiVec intrinsic
         vec_xst(acc_vec, 0, xacc + 2 * i);
         ^
In file included from xxhash.c:43:0:
xxhash.h: In function 'XXH3_accumulate_512_vsx':
xxhash.h:4148:9: error: invalid parameter combination for AltiVec intrinsic
         xxh_u64x2 acc_vec        = vec_xl(0, xacc + 2 * i);
         ^
xxhash.h:4158:9: error: invalid parameter combination for AltiVec intrinsic
         vec_xst(acc_vec, 0, xacc + 2 * i);
         ^
make: *** [libxxhash.so.0.8.1] Error 1
make: *** Waiting for unfinished jobs....
make: *** [xxhash.o] Error 1
```
Full log: https://koji.fedoraproject.org/koji/taskinfo?taskID=79410933

This failure is related to the use of altivec functions vec_xl and vec_xsl.
RHEL 7 uses gcc 4.8.5.
xxhash 0.8.0 and earlier versions compile on RHEL7 ppc64le.
If I revert the commit b40bf8dc by @mpe that introduces the use of vec_xl and vec_xsl the compilation succeeds.

--- a/xxhash.h
+++ b/xxhash.h
@@ -4129,7 +4129,7 @@ XXH3_accumulate_512_vsx(  void* XXH_RESTRICT acc,
                     const void* XXH_RESTRICT secret)
 {
     /* presumed aligned */
-    unsigned long long* const xacc = (unsigned long long*) acc;
+    unsigned int* const xacc = (unsigned int*) acc;
     xxh_u64x2 const* const xinput   = (xxh_u64x2 const*) input;   /* no alignment restriction */
     xxh_u64x2 const* const xsecret  = (xxh_u64x2 const*) secret;    /* no alignment restriction */
     xxh_u64x2 const v32 = { 32, 32 };
@@ -4145,7 +4145,7 @@ XXH3_accumulate_512_vsx(  void* XXH_RESTRICT acc,
         /* product = ((xxh_u64x2)data_key & 0xFFFFFFFF) * ((xxh_u64x2)shuffled & 0xFFFFFFFF); */
         xxh_u64x2 const product  = XXH_vec_mulo((xxh_u32x4)data_key, shuffled);
         /* acc_vec = xacc[i]; */
-        xxh_u64x2 acc_vec        = vec_xl(0, xacc + 2 * i);
+        xxh_u64x2 acc_vec        = (xxh_u64x2)vec_xl(0, xacc + 4 * i);
         acc_vec += product;
 
         /* swap high and low halves */
@@ -4155,7 +4155,7 @@ XXH3_accumulate_512_vsx(  void* XXH_RESTRICT acc,
         acc_vec += vec_xxpermdi(data_vec, data_vec, 2);
 #endif
         /* xacc[i] = acc_vec; */
-        vec_xst(acc_vec, 0, xacc + 2 * i);
+        vec_xst((xxh_u32x4)acc_vec, 0, xacc + 4 * i);
     }
 }
 

